src/server/transports/rest/http_rest_transport_handler.ts

```diff
@@ -0,0 +1,609 @@
+  }
+
+  /**
+   * Converts REST MessageSendParams to internal format.
+   */
+  private fromRestMessageSendParams(rp: RestMessageSendParams): MessageSendParams {
+    return {
+      configuration: rp.configuration
+        ? {
+            acceptedOutputModes: rp.configuration.accepted_output_modes,
+            blocking: rp.configuration.blocking,
+            historyLength: rp.configuration.history_length,
+            pushNotificationConfig: rp.configuration.push_notification_config
+              ? {
+                  authentication: rp.configuration.push_notification_config.authentication
+                    ? {
+                        schemes: rp.configuration.push_notification_config.authentication.schemes,
+                        credentials:
+                          rp.configuration.push_notification_config.authentication.credentials,
+                      }
+                    : undefined,
+                  id: rp.configuration.push_notification_config.id,
+                  url: rp.configuration.push_notification_config.url,
+                }
+              : undefined,
+          }
+        : undefined,
+      message: this.fromRestMessage(rp.message),
+      metadata: rp.metadata,
+    };
+  }
+
+  /**
+   * Converts REST TaskPushNotificationConfig to internal format.
+   */
+  private fromRestTaskPushNotificationConfig(
+    rc: RestTaskPushNotificationConfig
+  ): TaskPushNotificationConfig {
+    return {
+      pushNotificationConfig: {
+        authentication: rc.push_notification_config.authentication
+          ? {
+              schemes: rc.push_notification_config.authentication.schemes,
+              credentials: rc.push_notification_config.authentication.credentials,
+            }
+          : undefined,
+        id: rc.push_notification_config.id,
+        url: rc.push_notification_config.url,
+      },
+      taskId: rc.task_id,
+    };
+  }
+
+  /**
+   * Converts internal TaskPushNotificationConfig to REST format.
+   */
+  private toRestTaskPushNotificationConfig(
+    c: TaskPushNotificationConfig
+  ): RestTaskPushNotificationConfig {
+    return {
+      push_notification_config: {
+        authentication: c.pushNotificationConfig.authentication
+          ? {
+              schemes: c.pushNotificationConfig.authentication.schemes,
+              credentials: c.pushNotificationConfig.authentication.credentials,
+            }
+          : undefined,
+        id: c.pushNotificationConfig.id,
+        url: c.pushNotificationConfig.url,
+      },
+      task_id: c.taskId,
+    };
+  }
+}
+
+// ============================================================================
+// Server-Sent Events (SSE) Support
+// ============================================================================
+
+// Re-export SSE utilities from shared module for convenience
+// Both JSON-RPC and REST transports use these utilities
+export { SSE_HEADERS, formatSSEEvent, formatSSEErrorEvent } from '../sse_utils.js';
```

Comment:
nit: I wouldn't do it, seems like we import them here only for reexport, it should be fine for express handler to use a shared sse_utils package explicitly (which is and should stay package internal).